### PR TITLE
Add source Wizzard component

### PR DESCRIPTION
**Pull request by rvsia**
# Sources

This package exports **Add Source Wizard** and **Add Source button**.
 
## Using

As a primary button

```jsx
import { AddSourceButton } from '@redhat-cloud-services/frontend-components-sources';


<AddSourceButton />
```

As a Wizard

```jsx
import { AddSourceWizard } from '@redhat-cloud-services/frontend-components-sources';


<AddSourceWizard isOpen={this.state.isOpen} onClose={ () => this.setIsOpen(false)/>
```

**Props**

| Prop        | Type           | Default  | Description |
| ------------- |:-------------:| :-----:| ------: |
| isOpen     | bool | false | You need to control yourselves if the wizard is open or not. (Not needed for the button version) |
| afterSuccess     | function | null | This function will be executed after closing the wizard successful finish step. In Sources-UI this method is used for updating the list of sources. |
| onClose     | function | null | This function will be executed after closing the wizard. Eg. set isOpen to false. |
| successfulMessage     | node | 'Your source has been successfully added.' | A message shown on the last page of the wizard. Can be customized when accessing from different app (eg. 'Source was added to Cost Management') |
| sourceTypes     | array | null | SourceTypes options. This prop can be used on pages, which have already loaded the source types, so there is no need to load them in this component. |

**Description**

- updates Add source wizard to PF4

## Examples:

Loading

Loading

Error submit

onError

Retry button (when error)

retryButton

Success submit

onSuccess

### CODE REVIEW

**Comment by rvsia** [comment]
@karelhala @Hyperkid123 @martinpovolny @terezanovotna It should be ready for review :pray: (Just need to resolve tests...)

**Comment by terezanovotna** [comment]
This looks awesome so far!! 
The loading state is not consistent with what the platform is doing, but this is [the only documentation](https://cloud.redhat.com/docs/storybook?path=/story/ux--loading-states) on loading I found.

File: packages/sources/src/addSourceWizard/index.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    state = {
+        isOpen: false
+    }
+
+    setIsOpen = (value) => this.setState({
+        isOpen: value
+    })
+
+    render() {
+        return <>
```
Comment (Hyperkid123) [inline packages/sources/src/addSourceWizard/index.js]: Can you not use the fragment short hand please? Its very easy to miss. :eyes:

Comment (rvsia) [reply packages/sources/src/addSourceWizard/index.js]: it's very easy to write :smiling_imp:

Comment (Hyperkid123) [reply packages/sources/src/addSourceWizard/index.js]: :no_good_man:

**Comment by rvsia** [comment]
@Hyperkid123 A lot of the code is just copied from Sources, I will focus on it in next PRs. Just need go step by step now. :steam_locomotive:

**Comment by Hyperkid123** [comment]
> @Hyperkid123 A lot of the code is just copied from Sources, I will focus on it in next PRs. Just need go step by step now.

I know just putting it on here so you won't forget :smirk:

**Comment by martinpovolny** [comment]
@rvsia : what about i18n? I see that there are strings that do not seem to be marked in any way. What's the plan?

**Comment by rvsia** [comment]
@martinpovolny yeah, I removed it because there is no way how to do it in this library. (Respectively I think there should be probably some shared way - some shared set of components in this library, some shared process how to collect texts, etc.) I didnt want to make this PR bigger than it is, however, we should discuss this with @karelhala and work on it as soon as possible.

**Comment by Hyperkid123** [comment]
> @rvsia : what about i18n? I see that there are strings that do not seem to be marked in any way. What's the plan?

@martinpovolny There is an official i18n package for chrome in preparation. All insights packages will use it eventually.

**Comment by rvsia** [comment]
> > @rvsia : what about i18n? I see that there are strings that do not seem to be marked in any way. What's the plan?
> 
> @martinpovolny There is an official i18n package for chrome in preparation. All insights packages will use it eventually.

Is there any PR or something about that to follow?

**Comment by Hyperkid123** [comment]
> > > @rvsia : what about i18n? I see that there are strings that do not seem to be marked in any way. What's the plan?
> > 
> > 
> > @martinpovolny There is an official i18n package for chrome in preparation. All insights packages will use it eventually.
> 
> @Hyperkid123
> Is there any PR or something about that to follow?

Not yet, there is an empty repo for now.

**Comment by karelhala** [comment]
Released in version [@redhat-cloud-services/frontend-components-sources/v/0.0.2](https://www.npmjs.com/package/@redhat-cloud-services/frontend-components-sources/v/0.0.2)

**Pull request merged**